scripts/Muon/GUI/Common/home_instrument_widget/home_instrument_widget_presenter.py

```diff
@@ -0,0 +1,217 @@
+
+    def update_view_from_model(self):
+        self.handle_loaded_first_good_data_checkState_change()
+        self.handle_loaded_time_zero_checkState_change()
+        self._view.set_instrument(self._model._data.instrument)
+
+    def clear_view(self):
+        self._view.set_time_zero(0.0)
+        self._view.set_first_good_data(0.0)
+        self._view.set_combo_boxes_to_default()
+        self._view.set_checkboxes_to_defualt()
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Time Zero
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def handle_user_changes_time_zero(self):
+        time_zero = self._view.get_time_zero()
+        self._model.set_user_time_zero(time_zero)
+
+    def handle_loaded_time_zero_checkState_change(self):
+        if self._view.time_zero_state():
+            time_zero = self._model.get_file_time_zero()
+            self._view.set_time_zero(time_zero)
+        else:
+            time_zero = self._model.get_user_time_zero()
+            self._view.set_time_zero(time_zero)
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # First Good Data
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def handle_user_changes_first_good_data(self):
+        first_good_data = self._view.get_first_good_data()
+        self._model.set_user_first_good_data(first_good_data)
+
+    def handle_loaded_first_good_data_checkState_change(self):
+        if self._view.first_good_data_state():
+            first_good_data = self._model.get_file_first_good_data()
+            self._view.set_first_good_data(first_good_data)
+        else:
+            first_good_data = self._model.get_user_first_good_data()
+            self._view.set_first_good_data(first_good_data)
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Rebin
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def handle_fixed_rebin_changed(self):
+        fixed_bin_size = float(self._view.get_fixed_bin_text())
+        self._model.add_fixed_binning(fixed_bin_size)
+
+    def handle_variable_rebin_changed(self):
+        variable_bin_size = float(self._view.get_fixed_bin_text())
+        self._model.add_variable_binning(variable_bin_size)
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Instrument
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def handle_instrument_changed(self):
+        """
+        User changes the selected instrument.
+        """
```

Comment:
Single line docstrings should be one-liners, see [here](https://www.python.org/dev/peps/pep-0257/).